### PR TITLE
Start pushing all images to quay.io/konflux-ci

### PR DESCRIPTION
**Pull request by chmeliik**
[STONEBLD-2339](https://issues.redhat.com//browse/STONEBLD-2339)

All the images published from this repo will now be pushed to `quay.io/konflux-ci/*`. Some will continue getting pushed to their old location.

| Old (continues getting updates?) | New |
| --- | --- |
| ✅ `redhat-appstudio-tekton-catalog/task-*` | `konflux-ci/tekton-catalog/task-*` |
| ✅ `redhat-appstudio-tekton-catalog/pipeline-*` | `konflux-ci/tekton-catalog/pipeline-*` |
| ✅ `redhat-appstudio-tekton-catalog/data-acceptable-bundles` | `konflux-ci/tekton-catalog/data-acceptable-bundles` |
| ❌ `redhat-appstudio-tekton-catalog/pull-request-builds` | `konflux-ci/pull-request-builds:(task\|pipeline)-*` |
| ❌ `redhat-appstudio/appstudio-utils` | `konflux-ci/appstudio-utils` |
| ❌ `redhat-appstudio/pull-request-builds:build-definitions-utils-*` | `konflux-ci/pull-request-builds:appstudio-utils-*` |

Also, `data-acceptable-bundles` will now contain bundle references for both `redhat-appstudio-tekton-catalog/task-*` and `konflux-ci/tekton-catalog/task-*`.

The full list of repos that we need created in <https://quay.io/konflux-ci>:

<details>
<summary>click me</summary>

```
tekton-catalog/task-acs-deploy-check
tekton-catalog/task-acs-image-check
tekton-catalog/task-acs-image-scan
tekton-catalog/task-apply-tags
tekton-catalog/task-buildah
tekton-catalog/task-buildah-10gb
tekton-catalog/task-buildah-20gb
tekton-catalog/task-buildah-24gb
tekton-catalog/task-buildah-6gb
tekton-catalog/task-buildah-8gb
tekton-catalog/task-buildah-oci-ta
tekton-catalog/task-buildah-remote
tekton-catalog/task-buildah-rhtap
tekton-catalog/task-build-image-manifest
tekton-catalog/task-clair-scan
tekton-catalog/task-clamav-scan
tekton-catalog/task-deprecated-image-check
tekton-catalog/task-download-sbom-from-url-in-attestation
tekton-catalog/task-ecosystem-cert-preflight-checks
tekton-catalog/task-fbc-related-image-check
tekton-catalog/task-fbc-validation
tekton-catalog/task-gather-deploy-images
tekton-catalog/task-generate-odcs-compose
tekton-catalog/task-git-clone
tekton-catalog/task-git-clone-oci-ta
tekton-catalog/task-init
tekton-catalog/task-inspect-image
tekton-catalog/task-prefetch-dependencies
tekton-catalog/task-prefetch-dependencies-oci-ta
tekton-catalog/task-provision-env-with-ephemeral-namespace
tekton-catalog/task-rpm-ostree
tekton-catalog/task-s2i-java
tekton-catalog/task-s2i-nodejs
tekton-catalog/task-sast-snyk-check
tekton-catalog/task-sast-snyk-check-oci-ta
tekton-catalog/task-sbom-json-check
tekton-catalog/task-show-sbom
tekton-catalog/task-show-sbom-rhdh
tekton-catalog/task-slack-webhook-notification
tekton-catalog/task-source-build
tekton-catalog/task-source-build-oci-ta
tekton-catalog/task-summary
tekton-catalog/task-tkn-bundle
tekton-catalog/task-update-deployment
tekton-catalog/task-update-infra-deployments
tekton-catalog/task-upload-sbom-to-trustification
tekton-catalog/task-verify-enterprise-contract
tekton-catalog/task-verify-signed-rpms

tekton-catalog/pipeline-docker-build
tekton-catalog/pipeline-docker-build-oci-ta
tekton-catalog/pipeline-enterprise-contract
tekton-catalog/pipeline-enterprise-contract-everything
tekton-catalog/pipeline-enterprise-contract-slsa3
tekton-catalog/pipeline-fbc-builder
tekton-catalog/pipeline-java-builder
tekton-catalog/pipeline-nodejs-builder
tekton-catalog/pipeline-tekton-bundle-builder
tekton-catalog/pipeline-core-services-docker-build

tekton-catalog/data-acceptable-bundles

appstudio-utils

pull-request-builds

```

</details>

See the individual commits for more details.

### CODE REVIEW

File: .tekton/push.yaml [hunk 1]
```diff
@@ -105,6 +121,43 @@ spec:
               secretName: redhat-appstudio-tekton-catalog-build-definitions-pull-secret
```
Comment (chmeliik) [inline .tekton/push.yaml:121]: TODO: create this one in the konflux-ci OCP namespace

Comment (chmeliik) [reply .tekton/push.yaml:122]: and the `infra-deployments-pr-creator` as well

**Comment by chmeliik** [comment]
Attempted to test as much of it as I could in https://github.com/chmeliik/build-definitions/pull/3

It pushes the task bundles to https://quay.io/acmiel-rhtap and https://quay.io/acmiel-test (pretend that they are redhat-appstudio-tekton-catalog and konflux-ci)

### Checked

#### Pipeline bundles in org X correctly contain task bundles from org X

```bash
$ tkn bundle list quay.io/acmiel-rhtap/pipeline-docker-build:devel -o json | jq '.spec.tasks[].taskRef.params[] | select(.name == "bundle").value' -r
quay.io/acmiel-rhtap/task-init:0.2@sha256:683a0c782fe8dd9c9d9a01214f704f63855ded3165c50a55b7a0d9426360110b
quay.io/acmiel-rhtap/task-git-clone:0.1@sha256:8e6c1699823fc1c9157fe353c66f593fd9d7634c05a940a5040799642c08dff3
...
```

```bash
$ tkn bundle list quay.io/acmiel-test/tekton-catalog/pipeline-docker-build:devel -o json | jq '.spec.tasks[].taskRef.params[] | select(.name == "bundle").value' -r
quay.io/acmiel-test/tekton-catalog/task-init:0.2@sha256:4f704bcd12ba5d03d793cfec3c16e972eb2d01d3e2dff99cd17166653f223bf7
quay.io/acmiel-test/tekton-catalog/task-git-clone:0.1@sha256:5fe30d6c1e9b6d3eff4e5949a49144c30dfe4591d62f1f50da2734a62ae3dcbe
...
```

#### Bundles released to org X have a different sha than those released to org Y, but the digest of the layer *inside* the bundle is the same

```bash
$ skopeo inspect --raw docker://quay.io/acmiel-rhtap/task-init@sha256:683a0c782fe8dd9c9d9a01214f704f63855ded3165c50a55b7a0d9426360110b | jq '.layers[0].digest'
"sha256:dae6b844df0ad991e50c88feb7f02514148895a55df27b2e82aa950d43e364f5"

$ skopeo inspect --raw docker://quay.io/acmiel-test/tekton-catalog/task-init@sha256:4f704bcd12ba5d03d793cfec3c16e972eb2d01d3e2dff99cd17166653f223bf7 | jq '.layers[0].digest'
"sha256:dae6b844df0ad991e50c88feb7f02514148895a55df27b2e82aa950d43e364f5"
```

(the mismatch of the overall bundle digest is caused by a different `created` timestamp in the config)

#### The data-acceptable-bundles image contains bundle refs for both orgs

TODO

#### The data-acceptable-bundles image content is the same in both orgs

TODO

**Comment by chmeliik** [comment]
Changes required to run the pipelines in the konflux-ci namespace: https://github.com/redhat-appstudio/infra-deployments/pull/3804

**Review by mmorhun — APPROVED** [review]

File: .tekton/push.yaml [hunk 2]
```diff
@@ -105,9 +122,47 @@ spec:
...
+          - name: source
+            workspace: workspace
+        taskSpec:
+          params:
+            - name: revision
+              type: string
+          steps:
+            - name: build-bundles
+              image: quay.io/konflux-ci/appstudio-utils:{{ revision }}
+              # per https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting
```
Comment (lcarva) [inline .tekton/push.yaml:142]: This is one of those comments that just keeps getting copied around. I think it provides very little value and we should stop propagating the noise. :smile:

File: hack/trusted-tasks-update.sh [hunk 3]
```diff
@@ -10,25 +10,48 @@
...
 set -o nounset
 set -o pipefail
 
 mapfile -td ' ' COLLECT < <(echo -n "${COLLECT:-git oci}")
-INPUT_IMAGE=${INPUT_IMAGE:-quay.io/redhat-appstudio-tekton-catalog/data-acceptable-bundles:latest}
+mapfile -td ' ' QUAY_NAMESPACES < <(
+    echo -n "${QUAY_NAMESPACES:-'redhat-appstudio-tekton-catalog konflux-ci/tekton-catalog'}"
+)
+
+INPUT_IMAGE=${INPUT_IMAGE:-quay.io/konflux-ci/tekton-catalog/data-acceptable-bundles:latest}
```
Comment (lcarva) [inline hack/trusted-tasks-update.sh:32]: So this image is a bit unusual. We don't build it from scratch every time. Instead, this image gets incremental updates, which means the existing data matters.

When we have a new location for this image, we should "seed" it with some relevant data so it contains the full set of expected data. There are different ways of doing this.

1. Change the input image to always be the existing one from `redhat-appstudio-tekton-catalog`. As a follow up PR, we can switch to the new location.
2. Copy the existing image from `redhat-appstudio-tekton-catalog` to `konflux-ci` just before merging this PR.

Thinking about this further, there's still room for deviation between the two images. An alternative is to simply add a skopeo copy command at the end of the script to copy the image from `redhat-appstudio-tekton-catalog` to `konflux-ci`. At some point in the future, we just update the script to just maintain the `konflux-ci` one.

WDYT?

Comment (chmeliik) [reply hack/trusted-tasks-update.sh:32]: > Copy the existing image from redhat-appstudio-tekton-catalog to konflux-ci just before merging this PR.

:+1: This was the plan

> Thinking about this further, there's still room for deviation between the two images. An alternative is to simply add a skopeo copy command at the end of the script to copy the image from redhat-appstudio-tekton-catalog to konflux-ci. At some point in the future, we just update the script to just maintain the konflux-ci one.

Good point. I don't think I can skopeo copy at the end of the script (the credentials for pushing to redhat-appstudio-tekton-catalog are in a separate secret, `tkn bundle push` didn't like multiple `quay.io/...` auths in a single file). But I can probably skopeo copy in a separate step in the pipeline

Comment (zregvart) [reply hack/trusted-tasks-update.sh:32]: Just to note that this script is not part of the CI, we call it as needed when we miss an update to make sure to include it, so we can merge it as is.

**Review by zregvart — APPROVED** [review]

File: hack/trusted-tasks-update.sh (hunk 3 above)
Comment (chmeliik) [reply hack/trusted-tasks-update.sh:32]: Oh, TBH I didn't realize that we were talking about this script. The same applies to .tekton/scripts/build-acceptable-bundles.sh, so I updated the push pipeline to just `skopeo copy` the bundle from redhat-appstudio-tekton-catalog to konflux-ci/tekton-catalog

File: .tekton/push.yaml (hunk 2 above)
Comment (chmeliik) [reply .tekton/push.yaml]: Removed

File: hack/trusted-tasks-update.sh (hunk 3 above)
Comment (chmeliik) [reply hack/trusted-tasks-update.sh:32]: re-tested, now the bundles are more obviously identical which is nice
* https://quay.io/repository/acmiel-test/tekton-catalog/data-acceptable-bundles?tab=tags&tag=latest
* https://quay.io/repository/acmiel-rhtap/data-acceptable-bundles?tab=tags&tag=latest

**Comment by chmeliik** [comment]
> ⚠️ Note to self: need to copy the data-acceptable-bundles image to konflux-ci/tekton-catalog just before merging)

Did that: https://quay.io/repository/konflux-ci/tekton-catalog/data-acceptable-bundles?tab=tags

**Pull request merged**